pkgs/tools/video/blackmagic-desktop-video/default.nix

```diff
@@ -73,7 +73,7 @@ stdenv.mkDerivation rec {
   '';
 
   postUnpack = ''
-    tar xf Blackmagic_Desktop_Video_Linux_${lib.versions.majorMinor version}/other/${stdenv.hostPlatform.uname.processor}/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}.tar.gz
+    tar xf Blackmagic_Desktop_Video_Linux_${lib.head (lib.splitString "a" version)}/other/${stdenv.hostPlatform.uname.processor}/desktopvideo-${version}-${stdenv.hostPlatform.uname.processor}.tar.gz
```

Comment:
hmmm dupe code

somewhat curious why you need to extract the same tarball in both files, and if it should be extracted maybe in a separate derivation?

Reply:
the same tarball is needed for both the kernel module as well as the userspace tooling. the kernel module doesn't _technically_ require the userspace tooling (though in practice, it needs a proprietary binary to run since kernel module 11.5).

the duplicate code is ugly, i agree, and extracting it in a derivation would probably make sense. I don't know how exactly we'd want to structure that (since it's code shared between the kernel module and the userspace tools).

I'd want to fix that in a separate PR though. I'm planning to clean up this derivation a bit and package all the userland tools instead of just the bare minimum, that would probably the correct place to fix this code duplication as well.

Reply:
I don't know if it is _that_ important to share a common unpacked src derivation I guess

Reply:
Resolving given the conclusion, I'd say.